src/include/ipc/probe.h

```diff
@@ -0,0 +1,134 @@
+/* SPDX-License-Identifier: BSD-3-Clause
+ *
+ * Copyright(c) 2019 Intel Corporation. All rights reserved.
+ *
+ * Author: Adrian Bonislawski <adrian.bonislawski@linux.intel.com>
+ *         Artur Kloniecki <arturx.kloniecki@linux.intel.com>
+ */
+
+/**
+ * \file include/ipc/probe.h
+ * \brief Probe IPC definitions
+ * \author Adrian Bonislawski <adrian.bonislawski@linux.intel.com>
+ * \author Artur Kloniecki <arturx.kloniecki@linux.intel.com>
+ */
+
+#ifndef __IPC_PROBE_H__
+#define __IPC_PROBE_H__
+
+#include <ipc/header.h>
+#include <sof/bit.h>
+#include <stdint.h>
+
+#define PROBE_PURPOSE_EXTRACTION	0x1
+#define PROBE_PURPOSE_INJECTION		0x2
+
+#define PROBE_EXTRACT_SYNC_WORD		0xBABEBEBA
+
+#define PROBE_SHIFT_FMT_TYPE		31
+#define PROBE_SHIFT_STANDARD_TYPE	27
+#define PROBE_SHIFT_AUDIO_FMT		23
+#define PROBE_SHIFT_SAMPLE_RATE		19
+#define PROBE_SHIFT_NB_CHANNELS		14
+#define PROBE_SHIFT_SAMPLE_SIZE		12
+#define PROBE_SHIFT_CONTAINER_SIZE	10
+#define PROBE_SHIFT_SAMPLE_FMT		9
+#define PROBE_SHIFT_SAMPLE_END		8
+#define PROBE_SHIFT_INTERLEAVING_ST	7
+
+#define PROBE_MASK_FMT_TYPE		MASK(31, 31)
+#define PROBE_MASK_STANDARD_TYPE	MASK(30, 27)
+#define PROBE_MASK_AUDIO_FMT		MASK(26, 23)
+#define PROBE_MASK_SAMPLE_RATE		MASK(22, 19)
+#define PROBE_MASK_NB_CHANNELS		MASK(18, 14)
+#define PROBE_MASK_SAMPLE_SIZE		MASK(13, 12)
+#define PROBE_MASK_CONTAINER_SIZE	MASK(11, 10)
+#define PROBE_MASK_SAMPLE_FMT		MASK(9, 9)
+#define PROBE_MASK_SAMPLE_END		MASK(8, 8)
+#define PROBE_MASK_INTERLEAVING_ST	MASK(7, 7)
+
+/**
+ * Header for data packets sent via compressed PCM from extraction probes
+ */
+struct probe_data_packet {
+	uint32_t sync_word;		/**< PROBE_EXTRACT_SYNC_WORD */
+	uint32_t buffer_id;		/**< Buffer ID from which data was extracted */
+	uint32_t format;		/**< Encoded data format */
```

Comment:
what are the possible values for this format? Above you also have FMT and AUDIO_FMT and SAMPLE_FMT, so you probably want an explanation on what this field represents

Reply:
Agreed. This format encoding was taken from cAVS, and honestly, it's a patchwork abomination. Some bits are used as bitmasks, some carry numerical values. I didn't want to flood the body of structure with elongated comments. Probably up top, where masks and shifts are defined, would be a good place to place that documentation. Do you agree with that?

Reply:
yes, I am not asking for the format to be explained in the comment but a pointer provided to a longer format description.
In general we cannot rely on non-public information for the probe IPC,  it has to be self-contained.

if the format is such a mess we should probably have an additional qualifier should we want to change that format at some point. Or use a reserved field for future cleanups.

Reply:
Description provided on top of definitions of shifts and masks.

Reply:
thanks, quite useful indeed